src/backend/executor/spi.c

```diff
@@ -1935,6 +1935,8 @@ _SPI_prepare_plan(const char *src, SPIPlanPtr plan)
 		List	   *stmt_list;
 		CachedPlanSource *plansource;
 
+		Assert(parsetree);
```

Comment:
This doesn't solve anything at all! I corrected my https://github.com/arenadata/gpdb/pull/732 and now it solves both of your problems.